c7n/resources/cloudfront.py

```diff
@@ -181,9 +179,64 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('mismatch-s3-owner')
+class MismatchS3Owner(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront,
+       and verify whether owner is different from Cloudfront account owner.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mismatch-s3-owner
+                resource: distribution
+                filters:
+                  - type: mismatch-s3-owner
+   """
+
+    schema = type_schema(
+        'mismatch-s3-owner')
+
+    permissions = ('s3:ListBuckets',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def is_s3_domain(self, domain_name):
+        if domain_name.endswith('.s3.amazonaws.com'):
```

Comment:
`return domain_name.endswith('.s3.amazonaws.com')`

Reply:
I didn't want to default to returning URL ending, in case we want to include other domain extensions in the future.

Reply:
what other domain extensions would there be for a filter called `mismatch-s3-owner`?

Reply:
I've seen some referenced as `s3-website-us-east-1.amazonaws.com"` but they're only listed with custom origins, so they seem to be static websites and S3 backed but may not necessarily be set up as Cloudfront distributions.

Reply:
per the docs it looks like the other cases are as follows:
```
bucket-name.s3-website-region.amazonaws.com
bucket-name.s3-website.region.amazonaws.com
```
source: https://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteEndpoints.html

also, looks like this is the actual list of s3 website endpoints: https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints

Unfortunately, it doesn't seem like there's a programmatic way to construct these urls

Reply:
this snippet will get you all the s3 endpoints but the hostnames != website domains

```python
import boto3
import botocore.loaders
import botocore.regions

client = boto3.client('ec2')
regions = client.describe_regions()['Regions']
regions = [x['RegionName'] for x in regions]

loader = botocore.loaders.create_loader()
data = loader.load_data('endpoints')

resolver = botocore.regions.EndpointResolver(data)

for r in regions:
    endpoint = resolver.construct_endpoint('s3', r)
    print endpoint['hostname']
```

Reply:
It seems like there isn't a `clean` approach to those other domains, since there's no programmatic way to construct them. I can just add the full endpoints list from https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints to a shared file, and point to that in identifying each domain. I thought about constructing the domain based on the region where Custodian is being run but that wouldn't work for buckets where the naming convention is based on a different region.

Reply:
The intent of this particular PR is addressing Cloudfront distributions so I'm comfortable with foregoing other potential domains in the meantime. Especially as there isn't a clear `best practice` on how we ought to implement it. If there's a need in the future, though, we can revisit and implement it at that point.

Reply:
this is a blocker, it only works with buckets in us-east-1 (and even then not all) , and is simple to fix with a regex. "s3.*?amazonaws\.com"